test/integration/targets/pids/tasks/main.yml

```diff
@@ -0,0 +1,51 @@
+# Test code for the pid module
+# Copyright 2018, Saranya sridharan
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- pip:
+    name: psutil
+    
+- name: "Checking the empty result" 
+  pids: 
+    name: "blahblah"
+  register: emptypids
+   
+- name: "Verify that the list of Process IDs (PIDs) returned is empty"
+  assert:
+    that:
+    - emptypids is not changed
+    - emptypids.pids == [] 
+    
+- name: "Picking a random process name"
+  command: echo "{{ 99999999 | random }}"
+  register: random_name
+
+- name: "finding the 'sleep' binary"
+  command: which sleep
+  register: find_sleep
+
+- name: "copying 'sleep' binary"
+  copy:
+    src: "{{ find_sleep.stdout }}"
+    dest: "{{ output_dir }}/{{ random_name.stdout }}"
+    mode: "u+rwx"
+
+- name: "Running the copy of 'sleep' binary"
+  command: "{{ output_dir }}/{{ random_name.stdout }} 100"
+  async: 100
+  poll: 0
+
+- name: "Finding the process ID(PIDs)"
+  command: python {{ playbook_dir }}/targets/pids/tasks/obtainpid.py "{{ random_name.stdout }}"
+  register: find_process
```

Comment:
This is effectively testing the same code as the module uses to get the pid. It might be a good idea to start the background sleep process using a bash or python script that writes the child pid to a file in the output_dir instead. That way we can check the module results against that instead.